apps/dashboard/src/contract-ui/tabs/tokens/components/airdrop-form.tsx

```diff
@@ -1,167 +1,144 @@
+            {airdropFormOpen ? (
+              <AirdropUploadERC20
+                onClose={() => setAirdropFormOpen(false)}
+                setAirdrop={(value) =>
+                  setValue("addresses", value, { shouldDirty: true })
+                }
+              />
+            ) : (
+              <div className="flex flex-col md:flex-row gap-4">
+                <Button
+                  colorScheme="primary"
+                  borderRadius="md"
+                  onClick={() => setAirdropFormOpen(true)}
+                  rightIcon={<Upload size={16} />}
+                >
+                  Upload addresses
+                </Button>
 
-                  <Flex
-                    gap={2}
-                    direction="row"
-                    align="center"
-                    justify="center"
-                    color={addresses.length === 0 ? "orange.500" : "green.500"}
-                  >
-                    {addresses.length > 0 && (
-                      <>
-                        <Icon as={BsCircleFill} boxSize={2} />
-                        <Text size="body.sm" color="inherit">
-                          <strong>{addresses.length} addresses</strong> ready to
-                          be airdropped
-                        </Text>
-                      </>
-                    )}
-                  </Flex>
-                </Flex>
-              </Stack>
-              <Text>
-                You can airdrop to a maximum of 250 addresses at a time. If you
-                have more, please do it in multiple transactions.
-              </Text>
-              <TransactionButton
-                transactionCount={1}
-                isLoading={sendTransaction.isPending}
-                type="submit"
-                colorScheme="primary"
-                disabled={!!address && addresses.length === 0}
-                alignSelf="flex-end"
-                isDisabled={!formState.isDirty}
-              >
-                Airdrop
-              </TransactionButton>
-            </Stack>
-          </form>
-        </Stack>
-      </DrawerBody>
+                {addresses.length > 0 && (
+                  <div className="gap-2 flex flex-row items-center justify-center text-green-500">
+                    <CircleCheck className="text-green-500" size={16} />
+                    <Text size="body.sm" color="inherit">
+                      <strong>{addresses.length} addresses</strong> ready to be
+                      airdropped
+                    </Text>
+                  </div>
+                )}
+              </div>
+            )}
+          </div>
+          {!airdropFormOpen && (
+            // Todo: for erc20 you definitely can airdrop to more than 250 recipients in 1 tx
+            // maybe tweak this number?
+            <Text>
+              You can airdrop to a maximum of 250 addresses at a time. If you
+              have more, please do it in multiple transactions.
+            </Text>
+          )}
```

Comment:
the way we should do this:
1. we know that a transfer takes 21k gas
2. multicall has a little bit of overhead
3. if we know the gas limit for the chain for a tx (do we?) -> divide by 21k -> split list into that many "batches" -> display "N batches" to user -> user clicks "send" and we pop N transaction windows until they are all sent

--- 

tbh this is better with engine :D

Reply:
I'm gonna make a separate Linear ticket for this one if you're okay with it

Reply:
Updated the text to make it more informative

Reply:
https://linear.app/thirdweb/issue/DASH-261/improve-ux-for-the-airdrop-form